Week7/app/src/main/java/com/kulloveth/covid19virustracker/data/Repository.kt

```diff
@@ -1,51 +1,55 @@
 package com.kulloveth.covid19virustracker.data
 
-import androidx.paging.LivePagedListBuilder
-import androidx.paging.PagedList
-import com.kulloveth.covid19virustracker.data.db.CovidDatabase
-import com.kulloveth.covid19virustracker.model.*
-
-class Repository(private val database: CovidDatabase) {
-
-
-    //fetch status from the internet
-    suspend fun fetchStatus():Result<List<CountryStatus>> = try {
-        val data = Injection.apiService.getStatusByCountry(200).data
-        Success(data.status)
-    }catch (error: Throwable) {
-        Failure(error)
-    }
-
-    //fetch news from the internet
-    suspend fun fetchNews():Result<List<Article>> = try{
-        val data = Injection.newsApiService.getCovidNews("COVID", API_KEY).articles
-        Success(data)
-    }catch (error:Throwable){
-        Failure(error)
+import androidx.work.*
+import com.kulloveth.covid19virustracker.App
+import com.kulloveth.covid19virustracker.worker.NOTIFICATION_CHANNEL_NAME
+import com.kulloveth.covid19virustracker.worker.NewsPeriodicWorker
+import com.kulloveth.covid19virustracker.worker.StatusPeriodicWorker
+import java.util.concurrent.TimeUnit
+
+private const val TAG_SYNC_DATA = "TAG_SYNC_DATA"
+private const val NEWS_TAG_SYNC_DATA = "NEWS_SYNC_DATA"
+
+class Repository{
+    // Create Network constraint
+    val constraints = Constraints.Builder()
+        .setRequiredNetworkType(NetworkType.CONNECTED)
+        .setRequiresBatteryNotLow(true)
+        .build()
+    val workManager = WorkManager.getInstance(App.getContext())
+
+    //build and enqeue status work
+    fun fetchStatus() {
```

Comment:
Nice work here, I'm not sure why the api wont give me information but the basic idea here is good.